core/trino-main/src/main/java/io/trino/sql/gen/ExpressionCompiler.java

```diff
@@ -81,7 +83,15 @@ public CacheStatsMBean getCursorProcessorCache()
 
     public Supplier<CursorProcessor> compileCursorProcessor(Optional<RowExpression> filter, List<? extends RowExpression> projections, Object uniqueKey)
     {
-        Class<? extends CursorProcessor> cursorProcessor = cursorProcessors.getUnchecked(new CacheKey(filter, projections, uniqueKey));
+        Class<? extends CursorProcessor> cursorProcessor;
+        try {
+            cursorProcessor = cursorProcessors.getUnchecked(new CacheKey(filter, projections, uniqueKey));
+        }
+        catch (UncheckedExecutionException e) {
+            throwIfInstanceOf(e.getCause(), TrinoException.class);
```

Comment:
I am not sure`TrinoException` should ever be raised here (https://github.com/trinodb/trino/issues/14677#issuecomment-1342495284)

What kind of `TrinoException` do you expect here? example?

Also, for the `concat(single_value)` problem, this should be handled by ExpressionAnalyzer.
let's have a test in `TestAnalyzer`

Reply:
Analyzer can't detect this argument mismatch because `concat` function takes variable arity. Other functions  throw TrinoException in `SqlScalarFunction#specialize` have this same issue. e.g. ArrayConcatFunction

Reply:
It looks like we need to improve our analysis capabilities then.
For example, `concat` could take two explicit arguments, and then varargs rest. It would make it invalid to call the function with one or zero argument(s).
Would it require a lot of code to write?

Alternatively, we can introduce a concept of minium and maximum arg count for varargs.

cc @martint @dain

Reply:
I agree with improving analysis logic, but verifying only the argument number is sufficient. For instance, MapToJsonCast verifies the argument types in `specialize` method.